src/vec.rs

```diff
@@ -0,0 +1,321 @@
+    /// values (same order as the VariableLabels in Desc). If that combination of
+    /// label values is accessed for the first time, a new Metric is created.
+    ///
+    /// It is possible to call this method without using the returned Metric to only
+    /// create the new Metric but leave it at its start value (e.g. a Summary or
+    /// Histogram without any observations). See also the SummaryVec example.
+    ///
+    /// Keeping the Metric for later use is possible (and should be considered if
+    /// performance is critical), but keep in mind that Reset, DeleteLabelValues and
+    /// Delete can be used to delete the Metric from the MetricVec. In that case, the
+    /// Metric will still exist, but it will not be exported anymore, even if a
+    /// Metric with the same label values is created later. See also the CounterVec
+    /// example.
+    ///
+    /// An error is returned if the number of label values is not the same as the
+    /// number of VariableLabels in Desc.
+    ///
+    /// Note that for more than one label value, this method is prone to mistakes
+    /// caused by an incorrect order of arguments. Consider GetMetricWith(Labels) as
+    /// an alternative to avoid that type of mistake. For higher label numbers, the
+    /// latter has a much more readable (albeit more verbose) syntax, but it comes
+    /// with a performance overhead (for creating and processing the Labels map).
+    pub fn get_metric_with_label_values(&self, vals: &[&str]) -> Result<T::Output> {
+        self.v.get_metric_with_label_values(vals)
+    }
+
+    /// `get_metric_with` returns the Metric for the given Labels map (the label names
+    /// must match those of the VariableLabels in Desc). If that label map is
+    /// accessed for the first time, a new Metric is created. Implications of
+    /// creating a Metric without using it and keeping the Metric for later use are
+    /// the same as for GetMetricWithLabelValues.
+    ///
+    /// An error is returned if the number and names of the Labels are inconsistent
+    /// with those of the VariableLabels in Desc.
+    ///
+    /// This method is used for the same purpose as
+    /// `get_metric_with_label_values`. See there for pros and cons of the two
+    /// methods.
+    pub fn get_metric_with(&self, labels: &HashMap<&str, &str>) -> Result<T::Output> {
+        self.v.get_metric_with(labels)
+    }
+
+    /// `with_label_values` works as `get_metric_with_label_values`, but panics if an error
+    /// occurs. The method allows neat syntax like:
+    ///     httpReqs.with_label_values("404", "POST").inc()
+    pub fn with_label_values(&self, vals: &[&str]) -> T::Output {
+        self.get_metric_with_label_values(vals).unwrap()
+    }
+
+    /// `with` works as `get_metric_with`, but panics if an error occurs. The method allows
+    /// neat syntax like:
+    ///     httpReqs.with(Labels{"status":"404", "method":"POST"}).inc()
```

Comment:
just copy from go comment. 
You can help me to support a label macro to support this syntax later :-)